### ISSUE TITLE
AttributeError: module 'urllib3.util.url' has no attribute 'PERCENT_RE'

### ISSUE THREAD

**Issue by botbahlul** [streamlink/streamlink#5324]
### Checklist

- [X] This is a bug report and not a different kind of issue
- [X] [I have read the contribution guidelines](https://github.com/streamlink/streamlink/blob/master/CONTRIBUTING.md#contributing-to-streamlink)
- [X] [I have checked the list of open and recently closed bug reports](https://github.com/streamlink/streamlink/issues?q=is%3Aissue+label%3A%22bug%22)
- [X] [I have checked the commit log of the master branch](https://github.com/streamlink/streamlink/commits/master)

### Streamlink version

Latest stable release

### Description

C:\Program Files\Python310\lib\site-packages\streamlink\plugin\api\http_session.py line 53 typo?

I was trying to run my script with python 3.10.4 in Windows 10 and got that error, but trying to run it with python 3.8.6 is working fine, why?

### Debug log

```text
File "C:\Program Files\Python310\lib\site-packages\streamlink\plugin\api\http_session.py", line 53, in Urllib3UtilUrlPercentReOverride
    _re_percent_encoding: Pattern = urllib3.util.url.PERCENT_RE  # type: ignore[attr-defined]
AttributeError: module 'urllib3.util.url' has no attribute 'PERCENT_RE'. Did you mean: '_PERCENT_RE'?
```

**Comment by bastimeyer** [comment]
Streamlink applies an override for a faulty url encoding in urllib3, as you can see here:
- https://github.com/streamlink/streamlink/blob/5.4.0/src/streamlink/plugin/api/http_session.py#L41-L62
- https://github.com/streamlink/streamlink/blob/5.4.0/tests/test_api_http_session.py#L12-L33

Apparently, urllib3 has renamed its `PERCENT_RE` constant to `_PERCENT_RE` in its `2.0.0` release:
https://github.com/urllib3/urllib3/blame/2.0.0/src/urllib3/util/url.py#L15

**On the 1.x branch, it's still `PERCENT_RE`:**
https://github.com/urllib3/urllib3/blame/1.26.15/src/urllib3/util/url.py#L17

**Streamlink does only support urllib3 1.x, so if you've installed urllib3 2.x in your python environment, then you're ignoring the dependency version constraints of streamlink and requests:**
- streamlink: https://github.com/streamlink/streamlink/blob/5.4.0/setup.cfg#L49-L50
- requests: https://github.com/psf/requests/blob/v2.30.0/setup.cfg#L10

Not a bug. Fix your Streamlink install / python environment.

**Comment by botbahlul** [comment]
ok, thanks

**Comment by bastimeyer** [comment]
Use a virtual environment if you have other packages in your python environment which depend on urllib3 2.x.
https://streamlink.github.io/install.html#virtual-environment

**Comment by bastimeyer** [comment]
So apparently `requests` has published version `2.30.0` with a bump to `urllib3` `<3`:
https://github.com/psf/requests/releases/tag/v2.30.0

I unfortunately looked at the wrong package metadata yesterday. The right one is of course here:
https://github.com/psf/requests/blame/v2.30.0/setup.py#L64

They added a changelog just now after tagging the release yesterday already. For some reason, pip didn't pick up the new version for me when I ran `pip install -U --upgrade-strategy=eager -r dev-requirements.txt -r docs-requirements.txt` yesterday after seeing this thread here.

----

This means that Streamlink has to fix the new version incompabilities now.

I've restricted urllib3 to `<2` for now in #5325, but I'm going to bump that to `<3` again with the necessary changes.

This is a bit annoying, because `requests` should've bumped its major version in order to stay compliant with the semantic versioning scheme.

Speaking of semver, urllib3 isn't using this scheme correctly either, with lots of deprecations being added in 2.0.0, while stuff will get removed/changed in a minor release later on:
https://urllib3.readthedocs.io/en/stable/v2-migration-guide.html

So what needs to fix now is the name of the `urllib3.utils.url._PATTERN_RE` constant which Streamlink overrides, and maybe some other stuff which I'm going to look at now (TLS stuff / network interface stuff).

Reopening for visibility. Once fixed, there will be a new release, since the current stable release is broken when not limiting `urllib3` to `<2`.
